Bluepill-runner/Bluepill-runner/BPApp.m

```diff
@@ -35,17 +35,19 @@ + (instancetype)BPAppWithAppBundlePath:(NSString *)path withExtraTestBundles:(NS
 
     NSMutableArray *xcTestFiles = [[NSMutableArray alloc] init];
     for (NSString *filename in allFiles) {
-        NSString *extension = [[filename pathExtension] lowercaseString];
-        if ([extension isEqualToString:@"xctest"]) {
-            NSString *bundle = [xcTestsPath stringByAppendingPathComponent:filename];
-            NSString *basename = [filename stringByDeletingPathExtension];
-            NSString *executable = [bundle stringByAppendingPathComponent:basename];
-
-            BPXCTestFile *xcTestFile = [BPXCTestFile BPXCTestFileFromExecutable:executable
-                                                                      withError:error];
-            if (!xcTestFile) return nil;
-
-            [xcTestFiles addObject:xcTestFile];
+        if (onlyBundlePath && [[onlyBundlePath lastPathComponent] isEqual: [filename lastPathComponent]]) {
```

Comment:
I could also add this condition to the `if` on line 40. The extension would be needlessly calculated but that's not a big hit and it will prevent the extra nesting.